test/lit/passes/vacuum-intrinsics.wast

```diff
@@ -0,0 +1,233 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+;; RUN: wasm-opt %s --vacuum -all -S -o - | filecheck %s
```

Comment:
Now we don't have any changes in Vacuum.cpp, then where do we do these optimizations?

Reply:
Just by seeing that the call has no side effects, vacuum is able to remove more things. It will remove a drop of anything without side effects, for example.